[Bug]: volume discount figures shown for referral program

### Problem encountered

we are publishing volume discount figures in`BUS_EVENT_TYPE_FEES_STATS_UPDATED` and api `GetFeesStats` when there is no active volume discount program

looking at the trade, we dont actually apply the discount which is expected

### Observed behaviour

```
{"id":"383-9","block":"7973B7D30A203B845A0A567AE4B4980C3D8A9346196BA9217516B016895AA8CF","type":"BUS_EVENT_TYPE_FEES_STATS_UPDATED","feesStats":{"market":"063c6bbc4c517e5f2dec2ae27c729b353fec205c729890f171c8c8b7ada5121d","asset":"b29a266a165511a26a2e89de37ca35db13a2446c64c133bcc15ed5b040046de9","epochSeq":"37","totalRewardsPaid":[{"party":"0a56b567af60c7b2375e9872cfbaf5625457372ace53d987fcb2fcd3678b4b88","amount":"182634000000000000000"}],"referrerRewardsGenerated":[{"referrer":"0a56b567af60c7b2375e9872cfbaf5625457372ace53d987fcb2fcd3678b4b88","generatedReward":[{"party":"0a56b567af60c7b2375e9872cfbaf5625457372ace53d987fcb2fcd3678b4b88","amount":"182634000000000000000"}]}],"refereesDiscountApplied":[{"party":"d9544a417fce1a4894bf960ffe42a16d828fdcb059d81548290c1e6e8840b826","amount":"61000000000000000000"}],"volumeDiscountApplied":[{"party":"d9544a417fce1a4894bf960ffe42a16d828fdcb059d81548290c1e6e8840b826","amount":"61000000000000000000"}],"totalMakerFeesReceived":[{"party":"e5b38d36b53e2501dca1c864a0e7a656521b2c5d0f182dfedd8f2d55761d5962","amount":"396804800000000000000"}],"makerFeesGenerated":[{"taker":"d9544a417fce1a4894bf960ffe42a16d828fdcb059d81548290c1e6e8840b826","makerFeesPaid":[{"party":"e5b38d36b53e2501dca1c864a0e7a656521b2c5d0f182dfedd8f2d55761d5962","amount":"396804800000000000000"}]}]},"version":1,"chainId":"testnet-001","txHash":"7973B7D30A203B845A0A567AE4B4980C3D8A9346196BA9217516B016895AA8CF"}
```


```
        volume_discount = response["feesStats"]["volumeDiscountApplied"]
>       assert len(volume_discount) == 0, f"did not expect volume discount {volume_discount}"
E       AssertionError: did not expect volume discount [{'party': 'd9544a417fce1a4894bf960ffe42a16d828fdcb059d81548290c1e6e8840b826', 'amount': '61000000000000000000'}]
E       assert 1 == 0
E        +  where 1 = len([{'amount': '61000000000000000000', 'party': 'd9544a417fce1a4894bf960ffe42a16d828fdcb059d81548290c1e6e8840b826'}])
```

### Expected behaviour

if there is no volume discount program, dont publish the figures in `BUS_EVENT_TYPE_FEES_STATS_UPDATED` and the `GetFeesStats` api

### Steps to reproduce

```bash
1. Run the system-test `test_fees_stats_endpoint` on branch `feat-3020`
2. Observe the error
```


### Software version

dev branch

### Failing test

test_fees_stats_endpoint

### Jenkins run

https://jenkins.vega.rocks/job/common/job/system-tests-wrapper/23822/testReport/junit/tests.API/rest_endpoints_test/test_fees_stats_endpoint/

### Configuration used

_No response_

### Relevant log output

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -326,6 +326,7 @@
 - [9797](https://github.com/vegaprotocol/vega/issues/9797) - Default pagination limits are not always correctly set.
 - [9813](https://github.com/vegaprotocol/vega/issues/9813) - Add validation for team scope vs individual scope
 - [8570](https://github.com/vegaprotocol/vega/issues/8570) - Make sure the filtering by `cmd_type` on block explorer doesn't result with a bad query planning.
+- [9801](https://github.com/vegaprotocol/vega/issues/9801) - Set right numbers on volume discount in fees stats
 
 ## 0.72.1
 

--- a/core/fee/engine_test.go
+++ b/core/fee/engine_test.go
@@ -25,6 +25,7 @@ import (
 	"code.vegaprotocol.io/vega/core/types"
 	"code.vegaprotocol.io/vega/libs/num"
 	"code.vegaprotocol.io/vega/logging"
+	eventspb "code.vegaprotocol.io/vega/protos/vega/events/v1"
 
 	"github.com/golang/mock/gomock"
 	"github.com/stretchr/testify/assert"
@@ -67,7 +68,7 @@ func TestFeeEngine(t *testing.T) {
 	t.Run("calculate continuous trading fee", testCalcContinuousTrading)
 	t.Run("calculate continuous trading fee + check amounts", testCalcContinuousTradingAndCheckAmounts)
 
-	t.Run("calculate continuous trading fee + check amounts with discounts and rewards", testCalcContinuousTradingAndCheckAmounts)
+	t.Run("calculate continuous trading fee + check amounts with discounts and rewards", testCalcContinuousTradingAndCheckAmountsWithDiscount)
 
 	t.Run("calculate continuous trading fee empty trade", testCalcContinuousTradingErrorEmptyTrade)
 	t.Run("calculate auction trading fee empty trade", testCalcAuctionTradingErrorEmptyTrade)
@@ -145,13 +146,13 @@ func testCalcContinuousTradingAndCheckAmounts(t *testing.T) {
 	discountRewardService.EXPECT().RewardsFactorMultiplierAppliedForParty(gomock.Any()).Return(num.DecimalZero()).AnyTimes()
 	volumeDiscountService.EXPECT().VolumeDiscountFactorForParty(gomock.Any()).Return(num.DecimalZero()).AnyTimes()
 	discountRewardService.EXPECT().GetReferrer(gomock.Any()).Return(types.PartyID(""), errors.New("not a referrer")).AnyTimes()
-	eng.UpdateFeeFactors(types.Fees{
+	require.NoError(t, eng.UpdateFeeFactors(types.Fees{
 		Factors: &types.FeeFactors{
 			MakerFee:          num.DecimalFromFloat(.000250),
 			InfrastructureFee: num.DecimalFromFloat(0.0005),
 			LiquidityFee:      num.DecimalFromFloat(0.001),
 		},
-	})
+	}))
 	trades := []*types.Trade{
 		{
 			Aggressor: types.SideSell,
@@ -190,6 +191,134 @@ func testCalcContinuousTradingAndCheckAmounts(t *testing.T) {
 	assert.Equal(t, infra, 1)
 	assert.Equal(t, recv, len(trades))
 	assert.Equal(t, pay, len(trades))
+	assert.Equal(t, &eventspb.FeesStats{
+		Market:                   "",
+		Asset:                    testAsset,
+		EpochSeq:                 0,
+		TotalRewardsPaid:         []*eventspb.PartyAmount{},
+		ReferrerRewardsGenerated: []*eventspb.ReferrerRewardsGenerated{},
+		RefereesDiscountApplied: []*eventspb.PartyAmount{
+			{
+				Party:  "party1",
+				Amount: "0",
+			},
+		},
+		VolumeDiscountApplied: []*eventspb.PartyAmount{
+			{
+				Party:  "party1",
+				Amount: "0",
+			},
+		},
+		TotalMakerFeesReceived: []*eventspb.PartyAmount{
+			{
+				Party:  "party2",
+				Amount: "125",
+			},
+		},
+		MakerFeesGenerated: []*eventspb.MakerFeesGenerated{
+			{
+				Taker: "party1",
+				MakerFeesPaid: []*eventspb.PartyAmount{
+					{
+						Party:  "party2",
+						Amount: "125",
+					},
+				},
+			},
+		},
+	}, eng.GetFeesStatsOnEpochEnd())
+}
+
+func testCalcContinuousTradingAndCheckAmountsWithDiscount(t *testing.T) {
+	eng := getTestFee(t)
+	ctrl := gomock.NewController(t)
+	discountRewardService := mocks.NewMockReferralDiscountRewardService(ctrl)
+	volumeDiscountService := mocks.NewMockVolumeDiscountService(ctrl)
+	discountRewardService.EXPECT().ReferralDiscountFactorForParty(gomock.Any()).Return(num.DecimalFromFloat(0.3)).AnyTimes()
+	discountRewardService.EXPECT().RewardsFactorMultiplierAppliedForParty(gomock.Any()).Return(num.DecimalFromFloat(0.2)).AnyTimes()
+	volumeDiscountService.EXPECT().VolumeDiscountFactorForParty(gomock.Any()).Return(num.DecimalFromFloat(0.1)).AnyTimes()
+	discountRewardService.EXPECT().GetReferrer(gomock.Any()).Return(types.PartyID(""), errors.New("not a referrer")).AnyTimes()
+	require.NoError(t, eng.UpdateFeeFactors(types.Fees{
+		Factors: &types.FeeFactors{
+			MakerFee:          num.DecimalFromFloat(.000250),
+			InfrastructureFee: num.DecimalFromFloat(0.0005),
+			LiquidityFee:      num.DecimalFromFloat(0.001),
+		},
+	}))
+	trades := []*types.Trade{
+		{
+			Aggressor: types.SideSell,
+			Seller:    "party1",
+			Buyer:     "party2",
+			Size:      5,
+			Price:     num.NewUint(100000),
+		},
+	}
+
+	ft, err := eng.CalculateForContinuousMode(trades, discountRewardService, volumeDiscountService)
+	assert.NotNil(t, ft)
+	assert.Nil(t, err)
+	transfers := ft.Transfers()
+	var pay, recv, infra, liquidity int
+	for _, v := range transfers {
+		if v.Type == types.TransferTypeLiquidityFeePay {
+			liquidity++
+			assert.Equal(t, num.NewUint(252), v.Amount.Amount)
+		}
+		if v.Type == types.TransferTypeInfrastructureFeePay {
+			infra++
+			assert.Equal(t, num.NewUint(127), v.Amount.Amount)
+		}
+		if v.Type == types.TransferTypeMakerFeeReceive {
+			recv++
+			assert.Equal(t, num.NewUint(64), v.Amount.Amount)
+		}
+		if v.Type == types.TransferTypeMakerFeePay {
+			pay++
+			assert.Equal(t, num.NewUint(64), v.Amount.Amount)
+		}
+	}
+
+	assert.Equal(t, liquidity, 1)
+	assert.Equal(t, infra, 1)
+	assert.Equal(t, recv, len(trades))
+	assert.Equal(t, pay, len(trades))
+	assert.Equal(t, &eventspb.FeesStats{
+		Market:                   "",
+		Asset:                    testAsset,
+		EpochSeq:                 0,
+		TotalRewardsPaid:         []*eventspb.PartyAmount{},
+		ReferrerRewardsGenerated: []*eventspb.ReferrerRewardsGenerated{},
+		RefereesDiscountApplied: []*eventspb.PartyAmount{
+			{
+				Party:  "party1",
+				Amount: "262",
+			},
+		},
+		VolumeDiscountApplied: []*eventspb.PartyAmount{
+			{
+				Party:  "party1",
+				Amount: "60",
+			},
+		},
+		TotalMakerFeesReceived: []*eventspb.PartyAmount{
+			{
+				Party:  "party2",
+				Amount: "64",
+			},
+		},
+		MakerFeesGenerated: []*eventspb.MakerFeesGenerated{
+			{
+				Taker: "party1",
+				MakerFeesPaid: []*eventspb.PartyAmount{
+					{
+						Party:  "party2",
+						Amount: "64",
+					},
+				},
+			},
+		},
+	}, eng.GetFeesStatsOnEpochEnd())
 }
 
 func testCalcContinuousTradingAndCheckAmountsWithDiscountsAndRewardsBySide(t *testing.T, aggressorSide types.Side) {

--- a/core/fee/rebate_stats.go
+++ b/core/fee/rebate_stats.go
@@ -190,10 +190,10 @@ func (f *FeesStats) ToProto(asset string) *eventspb.FeesStats {
 		})
 	}
 
-	volumeDiscountAppliedParties := maps.Keys(f.RefereeDiscountApplied)
+	volumeDiscountAppliedParties := maps.Keys(f.VolumeDiscountApplied)
 	sort.Strings(volumeDiscountAppliedParties)
 	for _, party := range volumeDiscountAppliedParties {
-		amount := f.RefereeDiscountApplied[party]
+		amount := f.VolumeDiscountApplied[party]
 		fs.VolumeDiscountApplied = append(fs.VolumeDiscountApplied, &eventspb.PartyAmount{
 			Party:  party,
 			Amount: amount.String(),

